### ISSUE TITLE
Dropdown - Type Error: Cannot convert undefined or null to object

### ISSUE THREAD

**Issue by JBottomley** [primefaces/primereact#803]
[x] bug report
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://forum.primefaces.org/viewforum.php?f=57
```

**Current behavior**
Dropdown populates with the items from an axios call and will get filtered accordingly with an existing item. However when trying to select a new option onClick I get the TypeError: Cannot convert undefined or null to object.

PLEASE NOTE: This functionality worked in Primereact 2.0.0-rc1 and we updated to 3.1.0 this week.

**Expected behavior**
Will be able to select a new item onClick 

**Minimal reproduction of the problem with instructions**
My options are as populated via the follow:

>> Features >> _apiGetConditonAssessmentFeatures > response.data = [{"value":"ab426da45c7b4928a9797533837782b0","label":"null"},{"value":"7925c29fb06c42deb5a55f442340ed4a","label":"Ancient Semi Natural Woodland"},{"value":"e8317ff9516f4434959ffcd604c5b473","label":"Mixed Habitat Mosaic"},{"value":"b7a69e2303cb42469ebe96198067cbe9","label":"Planted Ancient Woodland Site"},{}]

I am receiving the following error:

Uncaught TypeError: Cannot convert undefined or null to object
    at Function.equalsByValue (ObjectUtils.js:42)
    at Function.equals (ObjectUtils.js:28)
    at Dropdown.findOptionIndex (Dropdown.js:418)
    at Dropdown.updateHiddenSelect (Dropdown.js:406)
    at Dropdown.selectItem (Dropdown.js:389)
    at Dropdown.onOptionClick (Dropdown.js:329)
    at DropdownItem.onClick (DropdownItem.js:57)


* **React version:**
16.6.3

* **PrimeReact version:**
3.1.0

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
Chrome
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
all

**Comment by JBottomley** [comment]
Please note: We have added the dropdown component from an old project that is using Primereact v2.0.0-rc1 and this is working fine

**Comment by yasithA** [comment]
I came across the same problem. Any update/workarounds for this?

**Comment by JBottomley** [comment]
We are now using an old version of the dropdown component taken from an existing project we have. That version of Primereact is 2.0.0-rc1 instead of 3.1.0

**Comment by cagataycivici** [comment]
I couldn't replicate without an example but added a potential fix, please try with 3.1.1 which will be released later today.
